utilities/rpp-unittests/SOW3_OCL/BatchPD_ocl_pln1.cpp

```diff
@@ -0,0 +1,850 @@
+        }
+        else
+            cout << "Unable to open file!";
+    }
+    else if ((ip_bitDepth == 2) || (ip_bitDepth == 4))
+    {
+        Rpp8u *outputTemp;
+        outputTemp = output;
+        Rpp32f *outputf32Temp;
+        outputf32Temp = outputf32;
+
+        if (outputFile.is_open())
+        {
+            for (int i = 0; i < oBufferSize; i++)
+            {
+                outputFile << *outputf32Temp << ",";
+                *outputTemp = (Rpp8u)RPPPIXELCHECK(*outputf32Temp * 255.0);
+                outputf32Temp++;
+                outputTemp++;
+            }
+            outputFile.close();
+        }
+        else
+            cout << "Unable to open file!";
+    }
+    else if ((ip_bitDepth == 5) || (ip_bitDepth == 6))
+    {
+        Rpp8u *outputTemp;
+        outputTemp = output;
+        Rpp8s *outputi8Temp;
+        outputi8Temp = outputi8;
+
+        if (outputFile.is_open())
+        {
+            for (int i = 0; i < oBufferSize; i++)
+            {
+                outputFile << (Rpp32s)*outputi8Temp << ",";
+                *outputTemp = (Rpp8u)RPPPIXELCHECK(((Rpp32s)*outputi8Temp) + 128);
+                outputi8Temp++;
+                outputTemp++;
+            }
+            outputFile.close();
+        }
+        else
+            cout << "Unable to open file!";
+    }
+    Rpp8u *outputCopy = (Rpp8u *)calloc(oBufferSize, sizeof(Rpp8u));
+
+    if (outputFormatToggle == 1)
+    {
+        for (int i = 0; i < noOfImages; i++)
+        {
+            for (int j = 0; j < maxDstHeight; j++)
+            {
+                for (int k = 0; k < maxDstWidth; k++)
+                {
+                    for (int c = 0; c < ip_channel; c++)
+                    {
+                        int dstpix = i * maxDstHeight * maxDstWidth * ip_channel + j * maxDstWidth * ip_channel + k * ip_channel + c;
+                        int srcpix = i * maxDstHeight * maxDstWidth * ip_channel + (j * maxDstWidth + k) + c * maxDstHeight * maxDstWidth;
+                        if (j == 0 && k == 0)
+                            cout << dstpix << " " << srcpix << endl;
+                        outputCopy[dstpix] = output[srcpix];
+                    }
+                }
+            }
+        }
+        output = outputCopy;
+    }
+
+    rppDestroyGPU(handle);
+
+    mkdir(dst, 0700);
+    strcat(dst, "/");
+    count = 0;
+    for (j = 0; j < noOfImages; j++)
+    {
+        int op_size = maxHeight * maxWidth * ip_channel;
+        Rpp8u *temp_output = (Rpp8u *)calloc(op_size, sizeof(Rpp8u));
+        for (i = 0; i < op_size; i++)
+        {
+            temp_output[i] = output[count];
+            count++;
+        }
+        char temp[1000];
+        strcpy(temp, dst);
+        strcat(temp, imageNames[j]);
+        Mat mat_op_image;
+        if (ip_channel == 3)
+        {
+            mat_op_image = Mat(maxHeight, maxWidth, CV_8UC3, temp_output);
+            imwrite(temp, mat_op_image);
+        }
+        if (ip_channel == 1)
```

Comment:
Codacy found an issue: [The comparison 'ip_channel == 1' is always true.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)